### ISSUE TITLE
Add all SVG properties

### ISSUE THREAD

**Issue by skrat** [CSSLint/csslint#283]
I think this might be useful as SVG is becoming more common. It should recognize all non-CSS2 SVG properties as specified here http://www.w3.org/TR/SVG/styling.html#SVGStylingProperties

I tried myself but adding them to known-properties.js and building, doesn't seem to work.

**Comment by janaspage** [comment]
May I ask why this issue was closed? The linter is still flagging the SVG property `fill`, which is common in CSS when using SVG iconography. Thanks!

**Comment by himedlooff** [comment]
:+1: SVG's are now promoted as best practice especially for icons.

**Comment by Frank3K** [comment]
+1

**Comment by diogosg** [comment]
+1

**Comment by rolebi** [comment]
+1

**Comment by tabalinas** [comment]
+1

**Comment by gssbzn** [comment]
+1

**Comment by Leikam** [comment]
+1

**Comment by toptalo** [comment]
+1

**Comment by qm3ster** [comment]
In the meantime, how do I make it ignore specifically the property 'fill'?

**Comment by shchukin** [comment]
+1

**Comment by kschaefe** [comment]
It just complained about `fill: transparent` in my CSS.  It doesn't know what transparent is.

**Comment by CBuntrock** [comment]
Will this be done? having many errors of unknown property since I use more svg now

**Comment by josezulu** [comment]
Can we please get an update on this? will SVG properties ever be added?

**Comment by josezulu** [comment]
CBuntrock: I had the same issues until I realised my csslint version was outdated. Updated it and check again.

**Comment by jmisur** [comment]
for anyone who struggles with this, add this rule to your gulpfile and set `"known-properties": false` in `.csslintrc`.
It reports all properties which were not found in `svgProperties` just like known-properties rule would.

```
// CSSLint doesn't support SVG properties, so we create our custom rule based on
// https://github.com/CSSLint/csslint/blob/master/src/rules/known-properties.js
csslint.addRule({
    // rule information
    id: "svg",
    name: "SVG properties",
    desc: "Adds support for SVG properties",
    url: "",
    browsers: "All",

    // initialization
    init: function(parser, reporter) {
        "use strict";
        const rule = this;
        const svgProperties = ['fill', 'fill-opacity', 'stroke', 'stroke-width', 'shape-rendering', 'text-anchor'];

        parser.addListener("property", function(event) {
            if (event.invalid) {
                const propertyName = event.property.toString().toLowerCase();
                if (svgProperties.indexOf(propertyName) < 0) {
                    reporter.report(event.invalid.message, event.line, event.col, rule);
                }
            }
        });
    }
});
```

**Comment by wkeese** [comment]
I think you can close this ticket; all the properties have been added, see https://github.com/CSSLint/parser-lib/blob/master/src/css/Properties.js.
